### PR TITLE
[FEAT] [Images] [9/N] Infer `Image` type for PIL images on ingress.

### PR DESCRIPTION
**Pull request by clarkzinzow**
This PR infers the `Image` type for PIL images provided to `Series.from_pylist()`, and therefore for `Table.from_pydict()` and `DataFrame.from_pydict()`.

### CODE REVIEW

File: src/python/series.rs [hunk 1]
```diff
@@ -312,3 +321,45 @@ impl From<PySeries> for series::Series {
...
     }
 }
+
+fn infer_daft_dtype_for_sequence(
+    vec_pyobj: &[PyObject],
+    py: pyo3::Python,
+) -> PyResult<Option<DataType>> {
+    let py_pil_image_type = py
+        .import("PIL.Image")
+        .and_then(|m| m.getattr(pyo3::intern!(py, "Image")))?;
```
Comment (jaychia) [inline src/python/series.rs:331]: Would this code fail gracefully if PIL isn't installed in the venv?

Comment (clarkzinzow) [reply src/python/series.rs:331]: Ah right, I had that as a TODO but forgot to add that!

Comment (clarkzinzow) [reply src/python/series.rs]: Fixed!

File: src/python/series.rs [hunk 2]
```diff
@@ -312,3 +321,47 @@ impl From<PySeries> for series::Series {
         item.series
     }
 }
+
+fn infer_daft_dtype_for_sequence(
+    vec_pyobj: &[PyObject],
+    py: pyo3::Python,
+) -> PyResult<Option<DataType>> {
+    let py_pil_image_type = py
+        .import("PIL.Image")
```
Comment (samster25) [inline src/python/series.rs:330]: this should also be interned

File: src/python/series.rs [hunk 3]
```diff
@@ -312,3 +321,47 @@ impl From<PySeries> for series::Series {
...
+        .and_then(|m| m.getattr(pyo3::intern!(py, "Image")));
+    let mut dtype: Option<DataType> = None;
+    for obj in vec_pyobj.iter() {
+        let obj = obj.as_ref(py);
+        if let Ok(pil_image_type) = py_pil_image_type {
+            if obj.is_instance(pil_image_type)? {
+                let mode_str = obj.getattr("mode")?.extract::<String>()?;
```
Comment (samster25) [inline src/python/series.rs:337]: interned

File: src/python/series.rs [hunk 4]
```diff
@@ -312,3 +321,47 @@ impl From<PySeries> for series::Series {
...
+                let mode = ImageMode::from_str(&mode_str)?;
```
Comment (samster25) [inline src/python/series.rs:338]: If we are banking on the PIL string matching our `from_str`. lets make a note of it in the ENUM docs

**Review by samster25 — APPROVED** [review]

**Pull request merged**